src/user.ts

```diff
@@ -0,0 +1,177 @@
+import { Empty, PartialMessage } from '@bufbuild/protobuf';
+import { PromiseClient } from '@connectrpc/connect';
+import GrpcConnect from './connect';
+import CoreClient from './core';
+import { UserService } from './pkg/grpc/scalekit/v1/users/users_connect';
+import { 
+  CreateUserRequest as ProtoCreateUserRequest, 
+  CreateUserResponse,
+  DeleteUserRequest,
+  GetUserRequest,
+  GetUserResponse,
+  ListUserRequest,
+  ListUserResponse,
+  UpdateUserRequest as ProtoUpdateUserRequest,
+  UpdateUserResponse,
+  AddUserRequest,
+  AddUserResponse,
+  User,
+  UpdateUser 
+} from './pkg/grpc/scalekit/v1/users/users_pb';
+import { CreateUserRequest, UpdateUserRequest } from './types/user';
+
+export default class UserClient {
+  private client: PromiseClient<typeof UserService>;
+
+  constructor(
+    private readonly grpcConnect: GrpcConnect,
+    private readonly coreClient: CoreClient
+  ) {
+    this.client = this.grpcConnect.createClient(UserService);
+  }
+
+  /**
+   * Create a new user in an organization
+   * @param {string} organizationId The organization id
+   * @param {CreateUserRequest} options The user creation options
+   * @returns {Promise<CreateUserResponse>} The created user
+   */
+  async createUser(organizationId: string, options: CreateUserRequest): Promise<CreateUserResponse> {
+    if (!organizationId) {
+      throw new Error('organizationId is required');
+    }
+    if (!options.email) {
+      throw new Error('email is required');
+    }
+
+    const user = new User({
+      email: options.email,
+      userProfile: {
+        firstName: options.userProfile?.firstName,
```

Comment:
We shouldnt include this property if not provided. Same for last name
```suggestion
        ...(options.userProfile?.firstName && { firstName: options.userProfile?.firstName }),
```